src/util/loaders/arrayLoader.ts

```diff
@@ -0,0 +1,35 @@
+import { Loader, loadField, LoadingOptions, UnionLoader, ValidationException } from '../internal'
+
+export class ArrayLoader implements Loader {
+  items: Loader[]
+
+  constructor (items: Loader[]) {
+    this.items = items
+  }
+
+  async load (doc: any, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<any> {
+    if (!Array.isArray(doc)) {
+      throw new ValidationException('Expected a list')
+    }
+    let r: any[] = []
+    const errors: ValidationException[] = []
+    for (var val of doc) {
```

Comment:
If `doc` is an array with more than one element, and we successfully load the field from the doc, then the `for` loop doesn't loop though the other elements. If it fails to load the field from the doc, it throws a `ValidationException` and never loops through the other elements.

Maybe we could update the `if` in the beginning of this function to check that it is an array and that it is not empty? Or replace the for loop with something else?

Reply:
This was a reeeaally bad scoping mistake on my end, thanks a lot for finding this... 🍀